tests/baselines/reference/unusedVariablesWithUnderscoreInBindingElement.errors.txt

```diff
@@ -0,0 +1,145 @@
+    }
+    
+    function t3() {
+        const [_a1, [[_b1, c1]], d1, _e1] = [1, [[2, 3]], 4, 5];
+        console.log(c1, d1);
+    
+        const [_a2, [[_b2, _c2]], _d2, _e2] = [1, [[2, 3]], 4, 5];
+    
+        const [a3, [[b3, c3]], d3, e3] = [1, [[2, 3]], 4, 5];
+               ~~
+!!! error TS6133: 'a3' is declared but its value is never read.
+                     ~~
+!!! error TS6133: 'b3' is declared but its value is never read.
+                         ~~
+!!! error TS6133: 'c3' is declared but its value is never read.
+                               ~~
+!!! error TS6133: 'd3' is declared but its value is never read.
+                                   ~~
+!!! error TS6133: 'e3' is declared but its value is never read.
+    }
+    
+    function t4() {
+        const { a1: _a1, b1 } = { a1: 1, b1: 1 };
+        console.log(b1);
+    
+        const { a2, b2: _b2 } = { a2: 1, b2: 1 };
+        console.log(a2);
+    
+        const { a3: _a3, b3: _b3 } = { a3: 1, b3: 1 };
+    }
+    
+    function t5() {
+        const { a1: _a1, b1 } = { a1: 1, b1: 1 };
+                         ~~
+!!! error TS6133: 'b1' is declared but its value is never read.
+        const { a2, b2: _b2 } = { a2: 1, b2: 1 };
+                ~~
+!!! error TS6133: 'a2' is declared but its value is never read.
+        const { a3, b3 } = { a3: 1, b3: 1 };
+              ~~~~~~~~~~
+!!! error TS6198: All destructured elements are unused.
+    }
+    
+    function t6() {
+        const {
+            a1: _a1,
+            b1: {
+                b11: {
+                    b111: _b111,
+                    b112
+                }
+            },
+            c1,
+            d1: _d1
+        } = { a1: 1, b1: { b11: { b111: 1, b112: 1 } }, c1: 1, d1: 1 };
+        console.log(b112, c1);
+    
+        const {
+            a2: _a2,
+            b2: {
+                b21: {
+                    b211: _b211, b212: _b212
+                }
+            },
+            c2: _c2,
+            d2: _d2
+        } = { a2: 1, b2: { b21: { b211: 1, b212: 1 } }, c2: 1, d2: 1 };
+    
+        const {
+            a3,
+            ~~
+!!! error TS6133: 'a3' is declared but its value is never read.
+            b3: {
+                b31: {
+                     ~
+                    b311, b312
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~
+                }
+    ~~~~~~~~~~~~~
+!!! error TS6198: All destructured elements are unused.
+            },
+            c3,
+            ~~
+!!! error TS6133: 'c3' is declared but its value is never read.
+            d3
+            ~~
+!!! error TS6133: 'd3' is declared but its value is never read.
+        } = { a3: 1, b3: { b31: { b311: 1, b312: 1 } }, c3: 1, d3: 1 };
+    }
+    
+    function t7() {
+        // error
+        const { _a1, _b1 } = { _a1: 1, _b1: 1 };
+              ~~~~~~~~~~~~
+!!! error TS6198: All destructured elements are unused.
+    
+        // ok
+        const { a2: _a2, b2: _b2 } = { a2: 1, b2: 1 };

```

Comment:
In retrospect this seems kind of questionable - why would you ever destructure and alias an unused object property?

Reply:
I don't really want to deal with "This is inconsistent with the array destructuring behavior!" bugs next. Seems fine.

Reply:
I agree, it's not worth touching again

Reply:
@DanielRosenwasser I agree that this is `questionable`, I can remove support for object destructuring and leave only array destructuring. What do you think?